### ISSUE TITLE
Implement inter-page annotations

### ISSUE THREAD

**Issue by mrtcode** [zotero/zotero#2761]
Logical annotation can often be divided between two pages. It makes sense to highlight both parts of it. But for that we need to extend `annotation.position`. It could look like this:
```
annotation.position.relatedPositions = [
  { pageIndex: 2, rects: […] }
];
```

`relatedAnnotation` won't be preserved when exporting to PDF, or it could be split into two. Although we'll still need to update it when using page deletion operation. Various changes in pdf-reader and pdf-worker are also necessary.

Page label will have page range for inter-page annotations.

User should be able to create inter-page annotation by starting selection in one page and finishing in the next page (no more than 2 pages in total). Although at the moment that wouldn't work well because header and footer is included. We have to implement structural PDF page analysing and exclude header and footer when selection crosses page boundary.

For now this could work only when holding SHIFT and selecting specific text.

**Comment by dstillman** [comment]
If it's only ever the next page, maybe we should simplify it?

```
annotation.position.nextPageRects = […];
```

(Or some similar same)

@michalrentka: What will happen if a new property appears in in `position`? Do we reject as unknown? If not, is it cleared on edit or kept? @mrtcode can comment, but probably it should be ignored and kept on edit, on both platforms. Blocking on unknown fields is obviously our standard approach, but probably fair to say that that's just at the level of the field (`annotationPosition`), and the whole point of a JSON blob is to give a bit more flexibility within it.

**Comment by michalrentka** [comment]
@dstillman I didn't check for unknown properties on this level. I only check whether there are all properties that I need from it and ignore the rest. So the annotation would be stored, but the new property `relatedPositions` would be ignored (not stored anywhere). When I implement this on iOS and the annotation is then edited, it would store and process the new property.

**Comment by dstillman** [comment]
Oh, so ignored means they wouldn't be stored? Meaning that if the user then modified the position in any way, the unknown property wouldn't be uploaded and would be lost.

I haven't checked, but I would imagine that an older desktop client would 1) save the whole `position` block (including the unknown property) and then 2) if updated, overwrite it without the new property.

This isn't the end of the world, but it means we should implement support for this on both platforms before actually rolling out support for creating it.

**Comment by michalrentka** [comment]
@dstillman that's right unfortunately. Not stored and not submitted if changed on iOS. I can store unknown properties and raw strings in the db so that I can submit them if something changes. I'll create a ticket for it on iOS repo.
